hms-dbmi-cellenics/api
# Caching hooks

# Description
Fixes:
* local env not using caching by default
* the marker heatmap and embeddings being pre-computed by hooks after the pipeline.

Can now run cache locally by using `USE_CACHE=true make run`. 
Documentation updated

**How to test in staging**
1. Run a pipeline for the first time
2. Check that the worker runs embedding & marker heatmap after the pipeline automatically, and save the ETags
3. Go to data exploration and open the devTools. 
4. Check that the 2 ETags precomputed by the pipeline appear in the network requests
5. Alternatively, when going to data exploration, make sure there are no new embeddings & marker heatmap requests (only list genes)

# Details
#### URL to issue
N/A

#### Link to staging deployment URL (or set N/A)
https://ui-ahriman-api24.scp-staging.biomage.net/

#### Links to any PRs or resources related to this PR


#### Integration test branch
master


# Merge checklist
Your changes will be ready for merging after all of the steps below have been completed.


### Code updates
Have best practices and ongoing refactors being observed in this PR
- [ ] All new dependency licenses have been checked for compatibility

### Manual/unit testing



### Integration testing
**You must check the box below** to run integration tests on the latest commit on your PR branch.
Integration tests have to pass before the PR can be merged. Without checking the box, your PR
**will not pass** the required status checks for merging.


### Documentation updates

### Optional
